salt/states/boto_ec2.py

```diff
@@ -452,3 +452,176 @@ def snapshot_created(name, ami_name, instance_name, wait_until_available=True, w
         sleep(5)
 
     return ret
+
+
+def instance_present(name, instance_name=None, instance_id=None, image_id=None,
+                     image_name=None, tags=None, key_name=None,
+                     security_groups=None, user_data=None, instance_type=None,
+                     placement=None, kernel_id=None, ramdisk_id=None,
+                     vpc_id=None, vpc_name=None, monitoring_enabled=None,
+                     subnet_id=None, subnet_name=None, private_ip_address=None,
+                     block_device_map=None, disable_api_termination=None,
+                     instance_initiated_shutdown_behavior=None,
+                     placement_group=None, client_token=None,
+                     security_group_ids=None, security_group_names=None,
+                     additional_info=None, tenancy=None,
+                     instance_profile_arn=None, instance_profile_name=None,
+                     ebs_optimized=None, network_interfaces=None,
+                     attributes=None, target_state=None, region=None, key=None,
+                     keyid=None, profile=None):
+    ### TODO - implement 'target_state={running, stopped}'
+    ### TODO - Describe all the above params :-(
```

Comment:
Would definitely be nice to have some basic arg docs before this hits a released version.